app/views/layouts/application.html.erb

```diff
@@ -21,6 +21,14 @@
           </button>
           <a class="navbar-brand" href="#">Foodies</a>
         </div>
+
+        <!-- default: search recipe -->
+        <%= form_tag(search_find_by_recipe_path, method: "get", class:"navbar-form navbar-left", role:"search") do%>
+        <div class="form-group">
+          <%= text_field_tag :search, params[:search], class: "form-control",placeholder: "Search.."%>
+          <% end %>
+        </div>
+
```

Comment:
The navbar-form introduced extra margins to the navigation bar. You could add the following lines in app/assets/stylesheets/style.scss (prabably around line # 258 )
```
.custom-nav .navbar-form {
    margin-top: 0;
    margin-bottom: 0;
}
```
Also, could you replace the tabs with spaces?